index.js

```diff
@@ -190,18 +192,15 @@ Client.prototype._pollConfig = function () {
         try {
           this.emit('config', JSON.parse(buf))
         } catch (e) {
-          res.destroy(e)
+          onError(e)
         }
       } else {
-        res.destroy(processConfigErrorResponse(res, buf))
+        onError(processConfigErrorResponse(res, buf))
       }
```

Comment:
I see here we've replaced a simple `res.destroy` call with a call to `onError`.  The old code will destroy the response object and do nothing else. The new code _does not_ destroy the response object, and also schedules a new configuration poll, and also emits an additional `request-error` event.  My naive assumption is we'd want to maintain the existing agent behavior -- can you talk a little bit about what you're trying to do here so we can figure out the best way to go about it?